src/index.ts

```
@@ -1,49 +1,148 @@
+
+    const message = formData.get("message");
+    if (typeof message !== "string") {
+      throw new Error(`request formData "message" is not a string`);
+    }
+    const signature = formData.get("signature");
+    if (typeof signature !== "string") {
+      throw new Error(`request formData "signature" is not a string`);
+    }
+
+    let siweMessage: SiweMessage;
+    try {
+      siweMessage = new SiweMessage(message);
+    } catch (err) {
+      // SiweMessage constructor throws a SiweError an object-type message is invalid
+      // https://github.com/spruceid/siwe/blob/23f7e17163ea15456b4afed3c28fb091b39feee3/packages/siwe/lib/client.ts#L352
+      if (err instanceof SiweError) {
+        return await this.failure(err.type, request, sessionStorage, options);
+      }
+
+      // SiweMessage constructor throws a generic Error if string message syntax is invalid
+      // https://github.com/spruceid/siwe/blob/23f7e17163ea15456b4afed3c28fb091b39feee3/packages/siwe-parser/lib/abnf.ts#L327
+      // the string message is parsed using the grammar defined in here:
+      // https://github.com/spruceid/siwe/blob/23f7e17163ea15456b4afed3c28fb091b39feee3/packages/siwe-parser/lib/abnf.ts#L23)
+      if (err instanceof Error) {
+        throw new Error(err.message);
+      }
+
+      throw new Error(
+        `SiweMessage constructor threw an unexpected error: ${String(err)}`
+      );
+    }
+    let user = {} as User;
+    await siweMessage
+      .verify(
+        {
+          signature,
+          domain: this.options.domain,
+        },
+        {
+          provider: this.options.provider,
+          suppressExceptions: true,
+        }
+      )
+      .then(async (siweResponse) => {
+        if (!siweResponse.success && !siweResponse.error) {
+          return await this.failure(
+            "siweResponse.success is false but no error was specified",
+            request,
+            sessionStorage,
+            options
+          );
```

Is throwing an error an acceptable thing to do in a remix-auth strategy? I'd throw an error here because this outcome indicates a programming error (not a user error).

---

I checked a couple of other strategies, I did not see them throwing generic errors, but `remix-auth` repo has an example of handling a [non-response errors](https://github.com/sergiodxa/remix-auth/blob/main/README.md?plain=1#L283) and that's where they fall. 
When I was testing on quest, it just threw the error without performing remix-auth `failureRedirect`. So I think it acceptable for a programming error 🤔